src/component/MovieList.js

```diff
@@ -48,16 +69,31 @@ const MovieList = () => {
     return (
         <div className="row">
             <div className="col-md-12">
-                <div className="search-container col-md-4">
-                    <input value={value} onChange={handleChange} placeholder="Search..." type="text"/>
+                <div className="row">
+                    <div className="search-container col-md-4">
+                        <input value={value} onChange={handleChange} placeholder="Search..." type="text" />
+                    </div>
+                    <div className="filter-container col-md-3">
+                        <DropdownButton
+                            menuAlign="right"
+                            title={filterValue}
+                            id="dropdown-menu-align-right"
+                        >
+                            {all.genres && all.genres.map((genre, i) => (
+                                <Dropdown.Item key={i} onClick={() => handleFilter(genre.id, genre.name)}>{genre.name}</Dropdown.Item>
+                            ))}
+                            <Dropdown.Divider />
+                            <Dropdown.Item onClick={() => handleClear()} style={{ color: 'red' }} eventKey="4">Clear</Dropdown.Item>
+                        </DropdownButton>
+                    </div>
                 </div>
                 <div className="col">
                     {all.data && all.data.map((movie, i) => (
-                        <MovieCard movie={movie} key={i}/>
+                        <MovieCard movie={movie} key={i} />
```

Comment:
Do we have a movie ID here? You can use this as `key`. Reference: https://reactjs.org/docs/lists-and-keys.html#keys